src/app/canary/canary.component.ts

```diff
@@ -0,0 +1,26 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+//  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+import { Component } from '@angular/core';
+import { GraphExplorerComponent } from '../GraphExplorerComponent';
+
+@Component({
+  selector: 'canary',
+  styleUrls: ['./canary.component.css'],
+  templateUrl: './canary.component.html',
+})
+export class CanaryComponent extends GraphExplorerComponent {
+
+  constructor() {
+    super();
+  }
+
+  public disableCanary() {
+    localStorage.setItem('GRAPH_MODE', null);
+    localStorage.setItem('GRAPH_URL', 'https://graph.microsoft.com');
+    location.reload();
```

Comment:
Unless I use the change detector method. (which also does a reload). 
A reload seems to be the only way the changes to the localstorage are propagated throughout

Reply:
Oh, that's because we load the URL into the UI on page load. Well, couldn't we still set it to localStorage and update the endpoint UI at the same time?

Or is couldn't we monitor the GRAPH_URL key in localStorage? This would end up in a lot of calls.

Reply:
This behaviour did not appear as a bug to me.

However, this is part the bigger state management problem in Graph Explorer. This looks like a reasonable hack.